src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,196 @@
+package parser;
+
+import command.DeleteCommand;
+import command.NotifyCommand;
+import command.SetBudgetCommand;
+import exceptions.NullException;
+import exceptions.InvalidCommand;
+import seedu.duke.FinancialGoal;
+import seedu.duke.Transaction;
+import seedu.duke.TransactionManager;
+import enumStructure.Category;
+import ui.Ui;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static constant.Constant.*;
+
+public class Parser {
+    /**
+     * Parses the user input and returns the corresponding command.
+     *
+     * @param userInput The raw user input string.
+     * @throws NullException If the input is invalid or missing required details.
+     */
+    public static void parser(String userInput, Ui ui, TransactionManager transactions, FinancialGoal goal) {
```

Comment:
This method is very long. Generally, keep the method length to less than 30 LOC.